packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -91,7 +95,7 @@ export function doStartSegmentCollection(
     { capture: true }
   )
 
-  function flushSegment(creationReason: CreationReason) {
+  function flushSegment(creationReason?: CreationReason) {
```

Comment:
what about using a `stop` reason?
even if it is not used, it could be more consistent

Reply:
I'm not fond of this, because `CreationReason` is exposed to web-ui. Adding unused variant would add some noise. I'll rework things a bit to be more explicit on what's going on, let me know if it'll look fine then.

Reply:
It is much clearer indeed